### PR TITLE
Make UUID path parameter conversion more flexible

### PR DESCRIPTION
**Pull request by edthrn**
# Summary

Based on @adriangb suggestion made in this [discussion thread](https://github.com/encode/starlette/discussions/2802#discussioncomment-11583842), here is a more flexible conversion of UUID path parameters. 

UUID representation may vary depending on the system from which it is generated. Some systems may return an hex representation with no hyphens, while some others use upper-case letters.

From Python perspective, they are all valid:
image

Not handling those cases may yield to hard-to-debug 404 errors.

# Checklist

- [x] I understand that this PR may be closed in case there was no previous discussion. (This doesn't apply to typos!)
- [x] I've added a test for each change that was introduced, and I tried as much as possible to make a single atomic change.

### CODE REVIEW

**Review by adriangb — APPROVED** [review]
looks good to me :)

File: tests/test_convertors.py [hunk 1]
```diff
@@ -58,15 +59,30 @@ def test_datetime_convertor(test_client_factory: TestClientFactory, app: Router)
...
+        ("uuid", "00000000aaaaffff9999000000000000", 200),
+        ("uuid", "00000000-AAAA-FFFF-9999-000000000000", 200),
+        ("uuid", "00000000AAAAFFFF9999000000000000", 200),
+        ("uuid", "not-a-uuid", 404),
+    ],
+)
+def test_default_convertor(
+    test_client_factory: TestClientFactory, param_type: str, param_value: str, status_code: int
+) -> None:
+    expected_types = {"float": float, "uuid": UUID}
```
Comment (Kludex) [inline tests/test_convertors.py]: Can we avoid this by leaving the `test_default_float_convertor` unchanged and create a new one for `uuid`? 🙏

**Review by Kludex — CHANGES_REQUESTED** [review]
This is fine. 🙏 

Just a minor nitpick, as I think our test suite focus a lot on readability.

File: tests/test_convertors.py (hunk 1 above)
Comment (edthrn) [reply tests/test_convertors.py]: Done :heavy_check_mark:

**Review by Kludex — APPROVED** [review]

**Comment by Kludex** [comment]
Thanks @edthrn 🙏

**Pull request merged**